pkg/apis/core/validation/validation.go

```diff
@@ -4352,6 +4352,11 @@ func ValidateService(service *core.Service) field.ErrorList {
 		}
 	}
 
+	// validate LoadBalancerClassName format
+	if service.Spec.LoadBalancerClassName != nil {
```

Comment:
you need to ensure this is only set if type = LB, right?

Reply:
my bad, I thought this was covered in `validateUpgradeDowngradeLoadBalancerClassName` but that's for update only.